subsys/bluetooth/mesh/op_agg.c

```diff
@@ -58,22 +54,23 @@ int bt_mesh_op_agg_send(struct bt_mesh_model *model,
 			struct bt_mesh_msg_ctx *ctx, struct net_buf_simple *msg,
 			const struct bt_mesh_send_cb *cb)
 {
-	uint16_t src;
 	int err;
 
 	/* Model responded so mark this message as acknowledged */
 	agg_ctx.ack = true;
 
+#ifdef CONFIG_BT_MESH_OP_AGG_CLI
 	/* Store source address so that Opcodes Aggregator Client can match
 	 * response with source model
 	 */
-	src = bt_mesh_model_elem(model)->addr;
+	uint16_t src = bt_mesh_model_elem(model)->addr;
```

Comment:
Hm... I'm not sure that opp code aggregator server should send frame without source address. Even more, comment above mentions that OpAgg client will use this field to figure our OpAgg server address.
That means server should add `src` address too.

Reply:
@omkar3141: What is your opinion about this? This mean including the srcs for the client and the server. It would be more similar to what it was before as OP_AGG_CLI and OP_AGG_SVR was defined on the same element in the tests.

Reply:
This `srcs` array is used internally (https://github.com/Balaklaka/zephyr/blob/cc1d531a1caf4be6e9b260b7e9f12f3c9f4730a2/subsys/bluetooth/mesh/op_agg_cli.c#L70) by the client to match responses with originating elements on the client side so that received Statuses can be dispatched to appropriate model instance. Spec does not prevent opcode aggragator client from combining messages originating from different element addresses on the client. However, the destination model instances for these messages must be on a single element on the server node identified by Element_Addresses filed of SEQENCE message. This is also clear from the place where [srcs](https://github.com/Balaklaka/zephyr/blob/cc1d531a1caf4be6e9b260b7e9f12f3c9f4730a2/subsys/bluetooth/mesh/op_agg.c#L28) is defined.

I don't see any issue here.

Reply:
@omkar3141, my concern was about `why its own address `src` is added into `srcs` array only for the client model and omitted for the server model`? 

What you wrote above didn't answer this question.

Reply:
> why its own address src is added into srcs array only for the client model and omitted for the server model

Because it is needed for the client model's functionality. This is not needed for server, because required address is specified within the SEQUENCE message (Element_Address field). 

There is no problem enabling it for server, but server's code will never make use of this array. It will simply consume a bit of RAM on server.

Reply:
thnx for clarification